driver-core/src/main/com/mongodb/internal/logging/StructuredLogger.java

```diff
@@ -82,22 +85,37 @@ public void log(final LogMessage logMessage) {
         if (interceptor != null) {
             interceptor.intercept(logMessage);
         }
-        //noinspection SwitchStatementWithTooFewBranches
         switch (logMessage.getLevel()) {
             case DEBUG:
-                if (logger.isDebugEnabled()) {
-                    LogMessage.UnstructuredLogMessage unstructuredLogMessage = logMessage.toUnstructuredLogMessage();
-                    String message = unstructuredLogMessage.interpolate();
-                    Throwable exception = logMessage.getException();
-                    if (exception == null) {
-                        logger.debug(message);
-                    } else {
-                        logger.debug(message, exception);
-                    }
-                }
+                logUnstructured(logMessage, Logger::isDebugEnabled, Logger::debug, Logger::debug);
```

Comment:
The inference magic is strong in this code, when it comes to the last two arguments.